lib/schemes/HttpUriPlugin.js

```diff
@@ -19,6 +19,34 @@ const memoize = require("../util/memoize");
 
 const getHttp = memoize(() => require("http"));
 const getHttps = memoize(() => require("https"));
+const proxyFetch = request => (url, options, callback) =>
+	new Promise(resolve => {
+		const key = "http_proxy";
+		const proxyUrl = process.env[key] || process.env[key.toUpperCase()];
+		if (!proxyUrl) return resolve();
+
+		const { hostname: host, port } = new URL(proxyUrl);
+
+		getHttp()
+			.request({
+				host, // IP address of proxy server
+				port, // port of proxy server
+				method: "CONNECT",
+				path: "raw.githubusercontent.com:80"
+			})
+			.on("connect", (res, socket) => {
+				if (res.statusCode === 200) {
+					// connected to proxy server
+					resolve(socket);
+				}
+			})
+			.on("error", err => {
+				console.error("Failed to connect to the proxy:", err);
+			})
+			.end();
+	}).then(socket =>
+		request.get(url, { ...options, ...(socket && { socket }) }, callback)
+	);
```

Comment:
I think will be better to create option for this, because setup can be different

Reply:
👍🏼  I'll commit it later, and I should check the test stages of CI at first because I failed test cases with timeout in my local environment.

```
 FAIL  test/ConfigTestCases.basictest.js (52.961 s, 316 MB heap size)
  ● ConfigTestCases › asset-modules › http-url › http-url should compile

    thrown: "Exceeded timeout of 30000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
```

Reply:
I think something locally with env, anyway option will be great here, so developer can use not only built-in `http`/`https`, but also any own solutions, including proxy

Reply:
@alexander-akait is it necessary to add a test case with a non-existed proxy to catch an error like this:

```
Error: Failed to connect to proxy server "http://127.0.0.1:12345/": connect ECONNREFUSED 127.0.0.1:12345
```